src/UID2.Client/UID2Encryption.cs

```diff
@@ -175,6 +186,16 @@ private static DecryptionResponse DecryptV2(byte[] encryptedId, KeyContainer key
             return new DecryptionResponse(DecryptionStatus.Success, idString, established, siteId, siteKey.SiteId, privacyBits.IsClientSideGenerated);
         }
 
+        private static bool IsValidDomainNameForSite(PrivacyBits privacyBits, int siteId, string domainName, KeyContainer keys)
+        {
+            if (!privacyBits.IsClientSideGenerated)
+            {
+                return true;
+            }
+
```

Comment:
if IsCSTG, can we have some test case if domainName is "", " " and even null to make sure they return false?

Reply:
Done